js-scripts/resolveS2Withdraw/resolveS2Withdraw.ts

```diff
@@ -0,0 +1,167 @@
+  const raw = await fs.readFile(CLAIMS_FILE, "utf8");
+  return JSON.parse(raw);
+};
+
+const readPoints = async (): Promise<PointMap> => {
+  const raw = await fs.readFile(CSV_FILE, "utf8");
+  const lines = raw
+    .trim()
+    .split("\n")
+    .filter((line) => line.length > 0);
+  lines.shift();
+  const entries = lines.map((line) => line.split(","));
+  return entries.reduce<PointMap>((acc, [address, points]) => {
+    if (!address || !points) {
+      return acc;
+    }
+    acc[canonical(address)] = Number(points);
+    return acc;
+  }, {});
+};
+
+const getProvider = () => {
+  const rpcUrl = process.env.MAINNET_RPC_URL;
+  if (!rpcUrl) {
+    throw new Error("Missing MAINNET_RPC_URL");
+  }
+  return new JsonRpcProvider(rpcUrl);
+};
+
+const withdrawInterface = new Interface([
+  "function withdraw(bool _claimRewards, address _receiver) external",
+  "function usersData(address user) external view returns (tuple(address rewardReceiver, address checkpointDelegatee, tuple(uint256 totalAccumulated, uint256 lastUpdate) stakeAgeInfo, uint256 effectiveBalance, tuple(uint256 amount, uint256 cooldownEnd) pendingWithdrawal))",
+]);
+
+const buildWithdrawTx = (safe: string, data: string) =>
+  RUMPEL_MODULE_INTERFACE.encodeFunctionData("exec", [
+    [
+      {
+        safe,
+        to: STAKED_RESOLV_ADDRESS,
+        data,
+        operation: 0,
+      },
+    ],
+  ]);
+
+const main = async () => {
+  const [claims, points] = await Promise.all([readClaims(), readPoints()]);
+  const provider = getProvider();
+  const contract = new Contract(
+    STAKED_RESOLV_ADDRESS,
+    withdrawInterface,
+    provider
+  );
+  const withdrawCalldata = withdrawInterface.encodeFunctionData(
+    "withdraw",
+    [false, RUMPEL_VAULT]
+  );
+
+  const latestBlock = await provider.getBlock("latest");
+  const now = latestBlock?.timestamp ?? 0;
+
+  let totalRewards = 0n;
+  let totalPoints = 0;
+
+  const rows: Array<Record<string, unknown>> = [];
+  const txs: Array<{ to: string; value: string; data: string }> = [];
+  const missingPoints: string[] = [];
+
+  for (const [safe, payload] of Object.entries(claims)) {
+    const userData = await contract.usersData(safe);
+    const pending = userData.pendingWithdrawal;
+    const withdrawalAmount = BigInt(pending.amount ?? 0n);
+    const cooldownEnd = Number(pending.cooldownEnd ?? 0n);
+
+    const claimAmount = BigInt(payload.minorAmount);
+    const amountMatchesClaim = withdrawalAmount === claimAmount;
+    if (!amountMatchesClaim) {
+      console.warn(
+        `Withdrawal amount mismatch for ${safe}: chain ${withdrawalAmount} vs claim ${claimAmount}`
+      );
+    }
+
+    const key = canonical(safe);
+    const hasPoints = Object.prototype.hasOwnProperty.call(points, key);
+    if (!hasPoints) {
+      missingPoints.push(safe);
+    }
+    const pointsForSafe = hasPoints ? points[key] : 0;
+
+    totalRewards += withdrawalAmount;
+    totalPoints += pointsForSafe;
```

Comment:
Since some point earners did not earn rewards, they are not represented in the claims object, and their point balances are not being included in the totalPoints. We allowed users without rewards to mint point tokens, so their balance should be factored into the redemption ratio. To calc total points, we should iterate on `points` and sum all balances. The result should only be slightly different given only very small point balances didn't earn rewards.